code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -27,10 +28,72 @@ var/const/MAX_ACTIVE_TIME = 400
 	var/strength = 5
 
 	var/attached = 0
+	var/target_time = 0.5 // seconds
+	var/walk_speed = 2 // tenths of seconds
+	var/nextwalk = 0
+	var/mob/living/carbon/human/target = null
 
 /obj/item/clothing/mask/facehugger/can_contaminate()
 	return 0
 
+/obj/item/clothing/mask/facehugger/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+
+/obj/item/clothing/mask/facehugger/process()
+	healthcheck()
+
+
+// HUGGER MOVEMENT AI
+/obj/item/clothing/mask/facehugger/proc/targetting()
+	spawn while(1)
```

Comment:
What is the point of have this be a processing object if this isn't under process()?

Also from the code here there's a chance this proc could prevent proper garbage collection since it doesn't seem like it would terminate even from being qdel()ed